paraflow-loader-consumer/src/main/java/cn/edu/ruc/iir/paraflow/loader/consumer/threads/ParquetFlushThread.java

```diff
@@ -1,22 +1,105 @@
+import cn.edu.ruc.iir.paraflow.metaserver.client.MetaClient;
+import cn.edu.ruc.iir.paraflow.metaserver.proto.MetaProto;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.parquet.column.ParquetProperties;
+import org.apache.parquet.example.data.Group;
+import org.apache.parquet.example.data.GroupFactory;
+import org.apache.parquet.example.data.simple.SimpleGroupFactory;
+import org.apache.parquet.hadoop.ParquetWriter;
+import org.apache.parquet.hadoop.example.GroupWriteSupport;
+import org.apache.parquet.schema.MessageType;
+import org.apache.parquet.schema.MessageTypeParser;
 
+import java.io.IOException;
 /**
  * paraflow
  *
  * @author guodong
  */
 public class ParquetFlushThread extends DataFlushThread
 {
+    private final MetaClient metaClient;
+    private Configuration conf = new Configuration();
+
     public ParquetFlushThread(String threadName)
     {
         super(threadName);
+        ConsumerConfig config = ConsumerConfig.INSTANCE();
+        metaClient = new MetaClient(config.getMetaServerHost(),
+                config.getMetaServerPort());
     }
 
     @Override
     boolean flushData(BufferSegment segment)
     {
-        return true;
+        String topic = segment.getFiberPartitions().get(0).getTopic();
+        int indexOfDot = topic.indexOf(".");
+        String dbName = topic.substring(0, indexOfDot);
+        int length = topic.length();
+        String tblName = topic.substring(indexOfDot + 1, length);
+        long beginTime = segment.getTimestamps()[0];
+        long endTime = segment.getTimestamps()[0];
+        for (long timeStamp : segment.getTimestamps()) {
+            if (timeStamp < beginTime) {
+                beginTime = timeStamp;
+            }
+            if (timeStamp > endTime) {
+                endTime = timeStamp;
+            }
+        }
+        String path = FileNameGenerator.generator(dbName, tblName, beginTime, endTime);
+        MetaProto.StringListType columnsNameList = metaClient.listColumns(dbName, tblName);
+        MetaProto.StringListType columnDataTypeList = metaClient.listColumnsDataType(dbName, tblName);
+        int columnNameCount = columnsNameList.getStrCount();
+        int columnDataTypeCount = columnDataTypeList.getStrCount();
+        System.out.println("columnNameCount : " + columnNameCount);
+        System.out.println("columnDataTypeCount : " + columnDataTypeCount);
+        if (columnNameCount == columnDataTypeCount) {
+            Path file = new Path(path);
+            try {
+                String schemaString = "message " + tblName + " {";
+                for (int i = 0; i < columnDataTypeCount; i++) {
+                    schemaString = schemaString + "required " + columnDataTypeList.getStr(i) + " " + columnsNameList.getStr(i) + "; ";
+                }
+                schemaString = schemaString + "}";
+                MessageType schema = MessageTypeParser.parseMessageType(schemaString);
```

Comment:
Please check Parquet supported types. I think it's limited. You need do some transformation before directly use our own type system